packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { PropTypes } from 'react';
+import ReactDatePicker from 'react-datepicker';
+import moment from 'moment';
+import ResponsiveElement from 'terra-responsive-element';
+import DateInput from './DateInput';
+import './DatePicker.scss';
+
+const propTypes = {
+  /**
+   * A moment object to use as the default end date for a date range.
+   */
+  endDate: PropTypes.object,
+  /**
+   * An array of moment objects that represent the dates to disable in the picker.
+   */
+  excludeDates: PropTypes.arrayOf(PropTypes.object),
+  /**
+   * A function that gets called for each date in the picker to evaluate which date should be disabled. A return value of true will be enabled and false will be disabled.
+   */
+  filterDate: PropTypes.func,
+  /**
+   * An array of moment objects that represent the dates to enable in the picker. All Other dates will be disabled.
+   */
+  includeDates: PropTypes.arrayOf(PropTypes.object),
+  /**
+   * Indicates the end date picker of a date range.
+   */
+  isEndDateRange: PropTypes.bool,
+  /**
+   * Indicates the start date picker of a date range.
+   */
+  isStartDateRange: PropTypes.bool,
+  /**
+   * A moment object to represents the maximum date that can be selected.
+   */
+  maxDate: PropTypes.object,
+  /**
+   * A moment object to represents the minimum date that can be selected.
+   */
+  minDate: PropTypes.object,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   */
+  onChange: PropTypes.func,
+  /**
+   * The selected date to show in the date input.
+   */
+  selectedDate: PropTypes.object,
+  /**
+   * A moment object to use as the default start date for a date range.
+   */
+  startDate: PropTypes.object,
+};
+
+const defaultProps = {
+  isEndDateRange: false,
+  isStartDateRange: false,
+};
+
+class DatePicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Added line breaks to clean up formatting. https://github.com/cerner/terra-core/commit/7870ad670ebe55044db8273441da4c5c5b8f95e5